lib/ui/painting.dart

```diff
@@ -1607,12 +1655,152 @@ class Path extends NativeFieldWrapperClass2 {
+
+  @override
+  PathMetric get current => _firstTime ? null : _pathMetric;
+
+  @override
+  bool moveNext() {
+    // PathMetric isn't a normal iterable - it's already initialized to its
+    // first Path.  Should only call _moveNext when done with the first one. 
+    if (_firstTime == true) {
+      _firstTime = false;
+      return true;
+    } else if (_pathMetric?._moveNext() == true) {
+      return true;
+    } else {
+      _pathMetric = null;
+      return false;
+    }
+  }
+}
+
+/// Utilities for measuring a [Path] and extracting subpaths.
+///
+/// Call [Path.computeMetrics] to create these objects. Once created, measures
+/// will only be valid while the path remains unmodified.  If the path is 
+/// modified, the behavior of the PathMetric object is undefined.
+class PathMetric extends NativeFieldWrapperClass2 {
+  /// Create a new empty [Path] object.
+  PathMetric._(Path path, bool forceClosed) { _constructor(path, forceClosed); }
+  void _constructor(Path path, bool forceClosed) native 'PathMeasure_constructor';
+
+  /// Return the total length of the current contour.
+  double get length native 'PathMeasure_getLength';
+
+  /// Pins distance to 0 <= distance <= getLength(), and then computes the 
+  /// corresponding position and tangent.
+  /// 
+  /// The [Tangent] object returns contains the position and angle (in radians)
+  /// of the tangent at specified distance. 
+  /// Returns null for a null or zero-length [Path].
+  Tangent getTangentForOffset(double distance) {
+    final Float32List posTan = _getPosTan(distance);
+    // first entry == 0 indicates that Skia returned false
+    if (posTan[0] == 0.0) {
+      return null;
+    } else {
+      return new Tangent(
+        new Offset(posTan[1], posTan[2]), 
+        math.atan2(posTan[4], posTan[3])
+      );
+    }
+  }
+  Float32List _getPosTan(double distance) native 'PathMeasure_getPosTan';
+
+  /// Given a start and stop distance, return the intervening segment(s).
+  /// 
+  /// startD and stopD are pinned to legal values (0..[length])
+  /// Returns null if the segment is 0 length or startD > stopD.
+  /// Begin the segment with a moveTo if startWithMoveTo is true.
+  Path extractPath(double startD, double endD, bool startWithMoveTo) native 'PathMeasure_getSegment';
+
+  /// Returns true if the path is closed.
+  bool get isClosed native 'PathMeasure_isClosed';
+
+  /// Move to the next contour in the path.
+  ///
+  /// A path can have a next contour if [Path.moveTo] was called after drawing began.
+  /// Return true if one exists, or false.
+  /// 
+  /// This is not exactly congruent with a regular [Iterator.moveNext].
+  /// Typically, [Iterator.moveNext] should be called before accessing the
+  /// [Iterator.current]. In this case, the [PathMetric] is valid before 
+  /// calling `_moveNext` - `_moveNext` should be called after the first
+  /// iteration is done instead of before.
+  bool _moveNext() native 'PathMeasure_nextContour';
```

Comment:
make these comments use `//` instead of `///` since this is private and those docs aren't "camera ready" (we'd want to rereview them before publishing them)